src/modules/jwt-encryption-service/JwtEncryptionService.ts

```diff
@@ -0,0 +1,54 @@
+import 'reflect-metadata';
+
+import { randomBytes } from 'crypto';
+import { JWE, JWK } from 'node-jose';
+
+export class JwtEncryptionService {
+
+    public generateNonce(): string {
+        const bytes = randomBytes(10);
+        const buffer = Buffer.from(bytes);
+        return buffer.toString('base64');
+    }
+
+    public async jweEncryptWithNonce(content: string, nonce: string, requestKey: string): Promise<string> {
```

Comment:
For me it makes more sense to combine `content: string, nonce: string` into one argument of object type. It gives more flexibility for it to be used in more generic use cases. Nounce is not relevant here, it is just payload.

After removing nounce from arg list please also remove `WithNounce` from name.

I'd generally keep it `encrypt` and `decrypt` cause class name defines what encryption it is.